src/TLC.sol

```diff
@@ -0,0 +1,308 @@
+        if (block.timestamp - positions[msg.sender].createdAt > interestRatePeriod || positions[msg.sender].createdAt == 0 ) {
+            positions[msg.sender].createdAt = block.timestamp;
+        }
+         
+        debtBalance -= amount;
+        IERC20(debtAddress).safeTransfer(
+            msg.sender,
+            amount
+        );
+        emit Borrow(msg.sender, amount);
+    }
+
+    function maxBorrowCapacity(address account) public returns(uint256) {
+        return ((positions[account].collateralAmount * collateralPrice * 100) / debtPrice / collateralizationRatio);
+    }
+
+
+
+   /**
+     * @dev Allows borrower to with draw collateral if sufficient to not default on loan
+     * @param withdrawalAmount is the amount to withdraw
+     */
+    function withdrawCollateral(uint256 withdrawalAmount) external {
+        if (withdrawalAmount > positions[msg.sender].collateralAmount) {
+            revert ExceededCollateralAmonut(msg.sender, positions[msg.sender].collateralAmount, withdrawalAmount);
+        }
+
+        uint256 maxBorrowCapacity = (((positions[msg.sender].collateralAmount - withdrawalAmount) * collateralPrice * 100) / debtPrice / collateralizationRatio);
+        if (positions[msg.sender].debtAmount > maxBorrowCapacity ) {
+            revert WillUnderCollaterlize(msg.sender, withdrawalAmount);
+        }
+
+        positions[msg.sender].collateralAmount -= withdrawalAmount;
+        collateralBalance -= withdrawalAmount;
+        IERC20(collateralAddress).safeTransfer(
+            msg.sender,
+            withdrawalAmount
+        );
+
+        emit Withdraw(msg.sender, withdrawalAmount);
+    }
+
+   /**
+     * @dev Allows borrower to repay borrowed amount
+     * @param repayAmount is the amount to repay
+     */
+    function repay(uint256 repayAmount) external {
+        if (repayAmount == 0) revert ZeroBalance(msg.sender);
+        positions[msg.sender].debtAmount = getTotalDebtAmount(msg.sender);
+
+        if (repayAmount >  positions[msg.sender].debtAmount) {
+            revert ExceededBorrowedAmount(msg.sender, positions[msg.sender].debtAmount, repayAmount);
+        }
+
+        positions[msg.sender].debtAmount -= repayAmount;
+        debtBalance += repayAmount;
+
+        uint256 periodsElapsed = (block.timestamp / interestRatePeriod) - (positions[msg.sender].createdAt / interestRatePeriod);
+        positions[msg.sender].createdAt += periodsElapsed * interestRatePeriod;
+        IERC20(debtAddress).safeTransferFrom(
+            msg.sender,
+            address(this),
+            repayAmount 
+        );
+        emit Repay(msg.sender, repayAmount);
+    }
+   
+   /**
+     * @dev Allows operator to liquidate debtors position
+     * @param debtor the account to liquidate
+     */
+    function liquidate(address debtor) external onlyOperators {
+
+        if (getCurrentCollaterilizationRatio(debtor) >= collateralizationRatio) {
```

Comment:
Could re-order and pass in the totalDebtOwed into this function so it doesn't have to calc it twice

Reply:
You can implement in terms of each other easily enough...

```
    function liquidate(address debtor) external onlyOperators {
        Position storage position = positions[debtor];    
        uint256 totalDebtOwed = getTotalDebtAmount(debtor);

        if (_getCurrentCollaterilizationRatio(position.collateralAmount, position.debtAmount, totalDebtOwed) >= minCollateralizationRatio) {
            revert OverCollaterilized(debtor);
        }

        if (collateralSeized > position.collateralAmount) {
           collateralSeized = position.collateralAmount;
        }

        position = Position({
            collateralAmount: position.collateralAmount - collateralSeized,
            debtAmount: 0,
            createdAt: 0
        });
        collateralToken.safeTransfer(
            debtCollector,
            collateralSeized 
        );

        emit Liquidated(debtor, totalDebtOwed, collateralSeized);
    }

    function _getCurrentCollaterilizationRatio(
        uint256 collateralAmount,
        uint256 debtAmount,
        uint256 totalDebtAmount
    ) internal view returns(uint256) {
        if (debtAmount == 0) {
            return 0;
        } else {
            return ((collateralAmount * collateralPrice * 100) / totalDebtAmount / debtPrice);
        }
    }

    function getCurrentCollaterilizationRatio(address account) public view returns(uint256) {
        return _getCurrentCollaterilizationRatio(positions[account].collateralAmount, positions[account].debtAmount, getTotalDebtAmount(account));
    }
```